pasta/base/token_generator.py

```diff
@@ -44,7 +44,7 @@ def __init__(self, source):
     self._parens = []
     self._hints = 0
     self._scope_stack = []
-    self._lines = source.splitlines(True)
+    self.lines = source.splitlines(True)
```

Comment:
i would move this to the top of the init to show the public things before the private